crates/ra_ide_api/src/completion/complete_postfix.rs

```diff
@@ -21,28 +22,46 @@ fn postfix_snippet(ctx: &CompletionContext, label: &str, detail: &str, snippet:
         .snippet_edit(edit)
 }
 
+fn is_bool_or_unknown(ctx: &CompletionContext) -> bool {
+    let receiver_ty = ctx.dot_receiver.and_then(|it| ctx.analyzer.type_of(ctx.db, it));
+    if let Some(ty) = receiver_ty {
+        match ty {
+            Ty::Apply(at) => match at.ctor {
+                TypeCtor::Bool => true,
+                _ => false,
+            },
+            Ty::Unknown => true,
+            _ => false,
+        }
+    } else {
+        true
+    }
+}
+
 pub(super) fn complete_postfix(acc: &mut Completions, ctx: &CompletionContext) {
     if let Some(dot_receiver) = ctx.dot_receiver {
         let receiver_text = dot_receiver.syntax().text().to_string();
-        postfix_snippet(ctx, "not", "!expr", &format!("!{}", receiver_text)).add_to(acc);
+        if is_bool_or_unknown(ctx) {
+            postfix_snippet(ctx, "not", "!expr", &format!("!{}", receiver_text)).add_to(acc);
```

Comment:
`not` could technically be overloaded on any type (although we don't handle that yet (although we could do it now))...

Reply:
Yeah, this looks like we'd lose the `std::ops::Not` impls on all integer types for example. Not sure if this is a big concern though.